lib/addons/p5.dom.js

```diff
@@ -788,55 +899,62 @@
   /**
    *
    * Sets the width and height of the element. AUTO can be used to
-   * only adjust one dimension.
+   * only adjust one dimension. If no arguments given returns the width and height
+   * of the element in an object.
    *
    * @method size
-   * @param  {Number} w width of the element
-   * @param  {Number} h height of the element
+   * @param  {Number} [w] width of the element
+   * @param  {Number} [h] height of the element
    * @return {p5.Element}
    */
   p5.Element.prototype.size = function(w, h) {
-    var aW = w;
-    var aH = h;
-    var AUTO = p5.prototype.AUTO;
-    if (aW !== AUTO || aH !== AUTO) {
-      if (aW === AUTO) {
-        aW = h * this.width / this.height;
-      } else if (aH === AUTO) {
-        aH = w * this.height / this.width;
-      }
-      // set diff for cnv vs normal div
-      if (this.elt instanceof HTMLCanvasElement) {
-        var j = {};
-        var k  = this.elt.getContext('2d');
-        for (var prop in k) {
-          j[prop] = k[prop];
+    if (arguments.length === 0){
+      return { 'width' : this.elt.offsetWidth , 'height' : this.elt.offsetHeight };
+    }else{
+      var aW = w;
+      var aH = h;
+      var AUTO = p5.prototype.AUTO;
+      if (aW !== AUTO || aH !== AUTO) {
+        if (aW === AUTO) {
+          aW = h * this.width / this.height;
+        } else if (aH === AUTO) {
+          aH = w * this.height / this.width;
         }
-        this.elt.setAttribute('width', aW * this._pInst.pixelDensity);
-        this.elt.setAttribute('height', aH * this._pInst.pixelDensity);
-        this.elt.setAttribute('style', 'width:' + aW + 'px !important; height:' + aH + 'px !important;');
-        this._pInst.scale(this._pInst.pixelDensity, this._pInst.pixelDensity);
-        for (var prop in j) {
-          this.elt.getContext('2d')[prop] = j[prop];
+        // set diff for cnv vs normal div
+        if (this.elt instanceof HTMLCanvasElement) {
+          var j = {};
+          var k  = this.elt.getContext('2d');
+          for (var prop in k) {
+            j[prop] = k[prop];
+          }
```

Comment:
It's a minor thing, but in general it's best to avoid defining variables (i.e., using `var`) inside a conditional block: these two functions are exactly the same in operation
```js
function one() {
  var a;
  if ( this.elt instanceof HTMLCanvasElement ) {
    a = true;
  }
}
// and
function one() {
  if ( this.elt instanceof HTMLCanvasElement ) {
    var a = true;
  }
}
```
and the former is usually considered preferable because it avoids the risk of unintuitive behavior due to [variable hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting).

Additionally, it would be best to avoid uninformative variable names like `j` and `k`: when p5 gets minified all variable names will be cleaned up anyway, so within the p5 code itself we try to use descriptive variable names like `autoWidth` or `2dDrawingContext`.